### ISSUE TITLE
Code Assist/Auto complete feature is not working for ReactJs

### ISSUE THREAD

**Issue by spitchaiah** [Microsoft/vscode#57444]
Issue Type: <b>Bug</b>

Hello,

In Visual Code editor, the code assist or Auto complete feature is not working  for ReactJS libraries.  However it works for my colleague running the same VS Code version & Mac OS. 

Eg:

```js
import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';

React.creatElement(

ReactDOM.render(element, document.getElementById('root'));
```

In the above example, when we type React.createElement, its not showing the method signature with the param names along with its type. Its the same with many of the React libraries.

Not sure if I'm missing something or this is a proxy issue. The above steps are from the the Code's tutorial on ReactJS:

https://code.visualstudio.com/docs/nodejs/reactjs-tutorial#_intellisense

Thanks for your help!

Regards,
Somu

VS Code version: Code 1.26.1 (493869ee8e8a846b0855873886fc79d480d342de, 2018-08-16T18:31:26.156Z)
OS version: Darwin x64 17.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz (8 x 2900)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.22GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron -psn_0_954601|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
xml|Dot|2.3.2
npm-intellisense|chr|1.3.0
path-intellisense|chr|1.4.2
vscode-eslint|dba|1.5.0
vscode-npm-script|eg2|0.3.5
search-node-modules|jas|1.3.0
react-vscode-extension-pack|jaw|0.3.1
sublime-babel-vscode|jos|0.2.10
vsliveshare|ms-|0.3.577
team|ms-|1.136.0
debugger-for-chrome|msj|4.8.2
java|red|0.29.0
vscode-java-debug|vsc|0.11.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.8.0
vscode-maven|vsc|0.10.0
vscode-react-native|vsm|0.6.16
JavaScriptSnippets|xab|1.7.0
ReactSnippets|xab|2.2.0

</details>

**Comment by mjbvz** [comment]
The intellisense for react in javascript comes from [automatic typings acquisition](https://code.visualstudio.com/Docs/languages/javascript#_automatic-type-acquisition).
To investigate why typing acquisition fails on your machine:

1. In your vscode setting, set `"typescript.tsserver.log": "verbose"`
1. Restart VS Code
1. Reproduce the issue
1. In VS Code with a js file open, run the `Open TS Server log` command
1. This should open a folder with a `ti-xxx.log` file in it. Please share that log file


To workaround this issue, you can also try manually installing these types using npm:

```
npm install --save-dev @types/react
```

**Comment by acdifran** [comment]
I had a similar issue and found it was due to the automatic typings acquisition process failing in the background. In the ti-xxx.log I was getting this error "env: node: No such file or directory" even though npm commands were running fine from my terminal. By default it seems VSCode runs this process using bash and I use zsh. The path for npm was not set up for default bash and so the npm commands were failing. I just had to export the path in bash_profile and now everything is working. Hopefully this helps for anyone else running into this.
